curl: wolfssl build without IPv6 is broken

Maintainer: @stangri 
Environment:
OpenWrt 21.02.2
mvebu-cortexa9-linksys_wrt3200acm

Description:
The 7.82 release of curl is broken when using wolfssl and IPv6 is disabled. It's patched in the latest curl source but not in the release.

Here is a patch for the 7.82 code:

```patch
--- curl-7.82.0.orig/lib/vtls/wolfssl.c
+++ curl-7.82.0.new/lib/vtls/wolfssl.c
@@ -462,9 +462,9 @@
     const char * const hostname = SSL_HOST_NAME();
     size_t hostname_len = strlen(hostname);
     if((hostname_len < USHRT_MAX) &&
-       (0 == Curl_inet_pton(AF_INET, hostname, &addr4)) &&
+       (0 == Curl_inet_pton(AF_INET, hostname, &addr4))
 #ifdef ENABLE_IPV6
-       (0 == Curl_inet_pton(AF_INET6, hostname, &addr6))
+       && (0 == Curl_inet_pton(AF_INET6, hostname, &addr6))
 #endif
       ) {
       size_t snilen;
```

log: 
OpenWrt-libtool: compile:  mips-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I../include -I../lib -I../lib -DBUILDING_LIBCURL -DCURL_HIDDEN_SYMBOLS -isystem /mnt/21g/21g/staging_dir/target-mips_24kc_musl/usr/include/wolfssl -isystem /mnt/21g/21g/staging_dir/toolchain-mips_24kc_gcc-8.4.0_musl/usr/include -isystem /mnt/21g/21g/staging_dir/toolchain-mips_24kc_gcc-8.4.0_musl/include/fortify -isystem /mnt/21g/21g/staging_dir/toolchain-mips_24kc_gcc-8.4.0_musl/include -DCURL_DISABLE_NTLM -isystem /mnt/21g/21g/staging_dir/target-mips_24kc_musl/usr/include -fvisibility=hidden -Os -pipe -mno-branch-likely -mips32r2 -mtune=24kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -fmacro-prefix-map=/mnt/21g/21g/build_dir/target-mips_24kc_musl/curl-7.82.0=curl-7.82.0 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -DPIC -fpic -ffunction-sections -fdata-sections -Werror-implicit-function-declaration -Wno-system-headers -MT vtls/libcurl_la-wolfssl.lo -MD -MP -MF vtls/.deps/libcurl_la-wolfssl.Tpo -c vtls/wolfssl.c  -fPIC -DPIC -o vtls/.libs/libcurl_la-wolfssl.o
vtls/wolfssl.c: In function 'wolfssl_connect_step1':
vtls/wolfssl.c:469:7: error: expected expression before ')' token
       ) {
       ^
vtls/wolfssl.c:471:57: error: 'snilen' undeclared (first use in this function); did you mean 'strlen'?
       char *snihost = Curl_ssl_snihost(data, hostname, &snilen);
                                                         ^~~~~~
                                                         strlen
vtls/wolfssl.c:471:57: note: each undeclared identifier is reported only once for each function it appears in
make[5]: *** [Makefile:2663: vtls/libcurl_la-wolfssl.lo] Error 1

--- a/net/curl/patches/0001-wolfssl-fix-compiler-error-without-IPv6.patch
+++ b/net/curl/patches/0001-wolfssl-fix-compiler-error-without-IPv6.patch
@@ -1,0 +1,26 @@
+From 471d5f44c5189bd78f8839ce8ac1b4b14f4b7d61 Mon Sep 17 00:00:00 2001
+From: Daniel Stenberg <daniel@haxx.se>
+Date: Mon, 7 Mar 2022 08:40:47 +0100
+Subject: [PATCH] wolfssl: fix compiler error without IPv6
+
+Reported-by: Joseph Chen
+Fixes #8550
+Closes #8552
+---
+ lib/vtls/wolfssl.c | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+--- a/lib/vtls/wolfssl.c
++++ b/lib/vtls/wolfssl.c
+@@ -462,9 +462,9 @@ wolfssl_connect_step1(struct Curl_easy *
+     const char * const hostname = SSL_HOST_NAME();
+     size_t hostname_len = strlen(hostname);
+     if((hostname_len < USHRT_MAX) &&
+-       (0 == Curl_inet_pton(AF_INET, hostname, &addr4)) &&
++       !Curl_inet_pton(AF_INET, hostname, &addr4)
+ #ifdef ENABLE_IPV6
+-       (0 == Curl_inet_pton(AF_INET6, hostname, &addr6))
++       && !Curl_inet_pton(AF_INET6, hostname, &addr6)
+ #endif
+       ) {
+       size_t snilen;

